chain-storage/src/permanent_store.rs

```diff
@@ -25,14 +24,12 @@ impl PermanentStore {
         let chain_length_index = data_pile::Database::file(chain_length_index_path)?;
 
         let root_id = root_id.into();
-        let id_length = root_id.as_ref().len();
 
         Ok(Self {
             blocks,
             chain_length_index,
             block_id_index,
             root_id,
-            id_length,
```

Comment:
I guess this could be used to avoid one in memory access if the database is of equal length elements, but I don't think datapile allows you to use the api in this way, so it's fine to remove